src/app/utils/query-parameter-sanitization.ts

```diff
@@ -0,0 +1,396 @@
+      value = sanitizeExpandQueryOptionValue(value);
+      break;
+    }
+
+    case '$filter': {
+      value = sanitizeFilterQueryOptionValue(value);
+      break;
+    }
+
+    default: {
+      if (!isAllAlpha(key)) {
+        key = '<invalid-key>';
+      }
+      value = '<value>';
+    }
+  }
+  return `${key}=${value}`;
+}
+
+/**
+ * @param queryOptionValue - The value of the $select query option is a
+ * comma-separated list of properties, qualified action names, qualified function names,
+ * the star operator (*), or the star operator prefixed with the namespace or alias of the schema.
+ * Examples:
+ * - GET /products?$select=rating,releaseDate
+ * - GET /products?$select=*
+ * - GET /products?$select=demoService.*
+ */
+function sanitizeSelectQueryOptionValue(queryOptionValue: string): string {
+  const selectedProperties = queryOptionValue.split(',');
+  selectedProperties.forEach((property, index) => {
+    property = property.trim();
+    if (!isAllAlpha(property) && property !== '*' && !property.endsWith('.*')) {
+      selectedProperties[index] = '<invalid-value>';
+    }
+  });
+
+  queryOptionValue = selectedProperties.join(',');
+  return queryOptionValue;
+}
+
+/**
+ * @param queryOptionValue - The value of the $format query option is a
+ * valid internet media type, optionally including parameters. Format-specific abbreviations may be used.
+ * Examples:
+ * - GET /orders?$format=application/json;metadata=full
+ * - GET /orders?$format=json
+ */
+function sanitizeFormatQueryOptionValue(queryOptionValue: string): string {
+  // Separate media type from parameters. Only one parameter is expected.
+  const formatSegments = queryOptionValue.split(';', 2);
+  const mediaType = formatSegments[0].trim();
+  queryOptionValue = !isMediaType(mediaType) ? '<invalid-media-type>' : mediaType;
+
+  // Check if there are parameters, key-value pairs
+  if (formatSegments.length > 1) {
+    let parameter = formatSegments[1].trim();
+    if (!isKeyValuePair(parameter)) {
+      parameter = '<invalid-parameter>';
+    }
+    queryOptionValue += `;${parameter}`;
+  }
+
+  return queryOptionValue;
+}
+
+/**
+ * @param queryOptionValue - The value of the $orderby query option is a
+ * a comma-separated list of expressions which includes a property name, a property path
+ * terminating on a primitive property and/or suffix asc/desc denoting direction of sorting
+ * Examples:
+ * - GET /users?orderby=displayName
+ * - GET /products?$orderby=releasedate asc,rating desc
+ * - GET /categories?$orderby=products/$count
+ */
+function sanitizeOrderByQueryOptionValue(queryOptionValue: string): string {
+  const sortingExpressions = queryOptionValue.split(',');
+
+  sortingExpressions.forEach((expr, index) => {
+    const expressionParts = expr.split(' ').filter(x => x !== ''); // i.e. property name and sort order
+    let propertyName = expressionParts[0].trim();
+    if (!isPropertyName(propertyName)) {
+      propertyName = '<invalid-value>';
```

Comment:
consider using `<invalid-property>` to distinguish from other `<invalid-value>`s